components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/ApiTypeWrapper.java

```diff
@@ -68,4 +68,68 @@ public void setTier(String tier) {
             api.getId().setTier(tier);
         }
     }
+
+    public void setContext(String context) {
+        if (isAPIProduct) {
+            apiProduct.setContext(context);
+        } else {
+            api.setContext(context);
+        }
+    }
+
+    public String getContext() {
+        if (isAPIProduct) {
+            return apiProduct.getContext();
+        } else {
+            return api.getContext();
+        }
+    }
+
+    public void setContextTemplate(String contextTemplate) {
+        if (isAPIProduct) {
+            apiProduct.setContextTemplate(contextTemplate);
+        } else {
+            api.setContextTemplate(contextTemplate);
+        }
+    }
+
+    public String getContextTemplate() {
+        if (isAPIProduct) {
+            return apiProduct.getContextTemplate();
+        } else {
+            return api.getContextTemplate();
+        }
```

Comment:
Shall we use a ternary operator here?
```suggestion
        return isAPIProduct ? apiProduct.getContextTemplate() : api.getContextTemplate();
```

Reply:
Sure